### PR TITLE
update exports field documentation

### PR DESCRIPTION
**Pull request by sokra**
add new node.js features for exports field
add support table for all general features
add rollup, esinstall and wmr to supported tools

### CODE REVIEW

**Comment by sokra** [comment]
@LarsDenBakker @lukastaegert I added rollup to the list of supported tools.

@matthewp @drwpow I added esinstall to the list of supported tools.

@developit I added wmr to the list of supported tools.

Could you review if I listed supported features and conditions for your tools correctly?

Here is a readable preview: https://webpack-js-org-git-content-exports-field.webpack-docs.now.sh/guides/package-exports/

**Comment by developit** [comment]
Hey @sokra - sent you a DM about thoughts on migrating this to somewhere central, but the issues you pointed out for WMR are all accurate. (yay, stuff to fix!)

**Comment by LarsDenBakker** [comment]
@sokra rollup supports "." property, normal property and property ending with / as well.

[@web/dev-server](https://www.npmjs.com/package/@web/dev-server) and [@web/test-runner](https://www.npmjs.com/package/@web/test-runner) both support the same features as rollup.

**Comment by matthewp** [comment]
@sokra I think that looks correct for esinstall. We're going to add support for some of this relatively soon, but I'll send a PR here if that happens after you've merged.

File: src/content/guides/package-exports.md [hunk 1]
```diff
@@ -32,18 +33,20 @@ An example:
...
-| `package/prefix/some/file.js` | `.../package/directory/some/file.js`  |
-| `package/prefix/deep/file.js` | `.../package/other-directory/file.js` |
-| `package/main.js`             | Error                                 |
+| Module request                      | Result                                           |
+| ----------------------------------- | ------------------------------------------------ |
+| `package`                           | `.../package/main.js`                            |
+| `package/sub/path`                  | `.../package/secondary.js`                       |
+| `package/prefix/some/file.js`       | `.../package/directory/some/file.js`             |
+| `package/prefix/deep/file.js`       | `.../package/other-directory/file.js`            |
+| `package/other-prefix/deep/file.js` | `.../package/yet-another/deep/file/deep/file.js` |
```
Comment (drwpow) [inline src/content/guides/package-exports.md:48]: This is honestly a great catch; I did not notice this behavior reading the Node docs but it is right there:

> All instances of `*` on the right hand side will then be replaced with this value

@matthewp do we test for this case in esinstall (and if not, we should)?

File: src/content/guides/package-exports.md [hunk 2]
```diff
@@ -232,17 +271,19 @@ Note: Since `production` and `development` is not supported by everyone, no assu
...
-| `browser`      | Code will run in a browser.     | webpack          |
-| `electron`     | Code will run in electron.      | webpack          |
-| `worker`       | Code will run in a (Web)Worker. | webpack          |
-| `worklet`      | Code will run in a Worklet.     |                  |
-| `node`         | Code will run in Node.js.       | webpack, Node.js |
-| `deno`         | Code will run in Deno.          |                  |
-| `react-native` | Code will run in react-native.  |                  |
+| Condition      | Description                                   | Supported by                        |
+| -------------- | --------------------------------------------- | ----------------------------------- |
+| `browser`      | Code will run in a browser.                   | webpack, esinstall, wmr             |
```
Comment (alexander-akait) [inline src/content/guides/package-exports.md:276]: /cc @chenxsan `esinstall`? Maybe `esbuild`?

Comment (chenxsan) [reply src/content/guides/package-exports.md:276]: I think they were talking about https://github.com/snowpackjs/snowpack/tree/main/esinstall#readme from snowpack.

**Pull request merged**